sql-plugin/src/main/scala/com/nvidia/spark/rapids/aggregate.scala

```diff
@@ -188,6 +213,31 @@ class GpuSortAggregateMeta(
             s"'complete', or 'all'")
       }
     }
+    hashAggReplaceMode match {
+      case "partial" => {
+        // In partial mode, if there are non distinct functions and multiple distinct functions
+        // non distinct functions are computed using First operator. The final result would be
+        // incorrect for non distinct functions for partition size > 1. Reason for this is - if the
+        // first batch computed and sent to CPU doesn't contain all the rows required to compute
+        // non distinct function(s), then Spark would consider that value as final result(due to
+        // First)Falling back to CPU for this special case.
+        if (agg.aggregateExpressions.exists(e => e.aggregateFunction.isInstanceOf[First])) {
+          val count = agg.aggregateExpressions.flatMap(expr =>
+            expr.children.flatMap {
+              _.collect {
+                case a: Count => a
+                case b: Count => b
```

Comment:
This should be average. Do we have a test for this added code path.

Reply:
Thanks! Good catch and previous test was not taking this path. Updated the test to do GpuSortAggregate.

Reply:
Why are we matching `hashAggReplaceMode` twice, cant we use the code above and add to it? Also ` _.collect `  I am not sure what performance implication this has. It goes over the tree, entirely or not, I dont know.

Reply:
Moved it to the above "partial" case.  
It goes over  each `AggregateExpression`  whose child is `Seq[Expression]` and creates a list based on the case statements.  I ran timings over this loop and runtime was negligible.  Please let me know if it's okay to have this partial function here OR do you want me to do further testing.